synapse/storage/_base.py

```diff
@@ -1592,6 +1598,9 @@ def _simple_select_list_paginate_txn(
         Args:
             txn : Transaction object
             table (str): the table name
+            filters (dict[str, T] | None):
```

Comment:
I kinda feel that once we're adding `LIKE` clauses to the query, this is quite specialised, and it would make more sense to give `get_users_paginate` its own `txn` function which can run the sql itself.

Reply:
Actually I guess some more endpoints with pagination could profit from this kind of query.

Reply:
also: can you give the parameters default values so that we don't need to list them all out in `_get_statistics_for_subject_txn` ?